back/config/config.go

```diff
@@ -13,115 +14,53 @@ import (
-	httpClient  httpclient.HttpApi
-	VaultClient vault.VaultClient
-}
-
-type HashicorpVault struct {
-	AppName        string `yaml:"app_name"`
-	AuthURL        string `yaml:"authentication-url"`
-	BaseURL        string `yaml:"base_url"`
-	OrganizationID string
-	ProjectID      string
-}
-
-type Text struct {
-	HashicorpVault HashicorpVault `yaml:"vault"`
-}
-
-type Secrets struct {
-	StripeId      string `yaml:"STRIPE_ID"`
-	StripeToken   string `yaml:"STRIPE_TOKEN"`
-	SendgridId    string `yaml:"SENGDRID_ID"`
-	SendgridToken string `yaml:"SENDGRID_TOKEN"`
-}
+func InitializeConfig() (Config, error) {
+	var config Config
 
-func InitializeConfig() (config Config, err error) {
 	envFile, err := os.ReadFile("config/default.yaml")
 	if err != nil {
-		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to read config file")
+		return Config{}, errors.Errorf("InitializeConfig, unable to read config file %s", err)
 	}
 
 	if err = yaml.Unmarshal(envFile, &config); err != nil {
-		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to unmarshal")
+		return Config{}, errors.Errorf("InitializeConfig, unable to unmarshall %s", err)
 	}
 
 	if err := godotenv.Load(); err != nil {
-		return Config{}, errors.Wrapf(err, "Couldn't load .env file")
+		return Config{}, errors.Errorf("Couldn't load .env file %s", err)
 	}
 
 	config.Handler, err = loadClientHandler(config)
 	if err != nil {
 		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to load client handler")
 	}
 
-	config, err = buildEnvironmentConfig(config)
+	initializedConfig, err := selectConfigProcessor[config.Env.BuildProduction](config)
 	if err != nil {
-		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to build environment config")
+		return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to build")
 	}
 
-	return
+	return initializedConfig, nil
 }
 
-func buildEnvironmentConfig(config Config) (Config, error) {
-
-	if !config.Env.BuildProduction {
-		config, err := loadConfig(config, stagingEnv)
-		if err != nil {
-			return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to load staging config")
-		}
-
-		err = InitDatabase(config, stagingEnv)
-		if err != nil {
-			return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to initialize staging database")
-		}
-
-		log.Info("Staging Environment has been loaded")
+func buildEnvironmentConfig(config Config, env string, configProcessor any) (Config, error) {
+	configFile, err := os.ReadFile(fmt.Sprintf("config/%s.yaml", env))
+	if err != nil {
+		return Config{}, errors.Errorf("loadConfig, unable to read config file %s", err)
+	}
 
-	} else {
-		config, err := loadConfig(config, productionEnv)
-		if err != nil {
-			return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to load config")
-		}
+	if err = yaml.Unmarshal(configFile, &configProcessor); err != nil {
+		return Config{}, errors.Errorf("loadConfig, unable to unmarshall config file %s", err)
```

Comment:
Shouldn't you use one of the unable you created in the constants you created, or create a new constant